tests/acceptance/features/bootstrap/SpacesContext.php

```diff
@@ -1710,6 +1729,97 @@ public function theUserHasCreatedASpaceByDefaultUsingTheGraphApi(
 		);
 	}
 
+	/**
+	 * @When /^user "([^"]*)" copies (?:file|folder) "([^"]*)" to "([^"]*)" in space "([^"]*)" using the WebDAV API$/
+	 *
+	 * @param string $user
+	 * @param string $fileSource
+	 * @param string $fileDestination
+	 * @param string $spaceName
+	 *
+	 * @return void
+	 * @throws GuzzleException
+	 */
+	public function userCopiesFileWithinSpaceUsingTheWebDAVAPI(
+		string $user,
+		string $fileSource,
+		string $fileDestination,
+		string $spaceName
+	):void {
+		$space = $this->getSpaceByName($user, $spaceName);
+		$headers['Destination'] = $this->destinationHeaderValueWithSpaceName(
+			$user,
+			$fileDestination,
+			$spaceName
+		);
+
+		$fullUrl = $space["root"]["webDavUrl"] . '/' . ltrim($fileSource, "/");
+		$this->copyFilesAndFoldersRequest($user, $fullUrl, $headers);
+	}
+
+	/**
+	 * @When /^user "([^"]*)" copies (?:file|folder) "([^"]*)" from space "([^"]*)" to "([^"]*)" inside space "([^"]*)" using the WebDAV API$/
+	 *
+	 * @param string $user
+	 * @param string $fileSource
+	 * @param string $fromSpaceName
+	 * @param string $fileDestination
+	 * @param string $toSpaceName
+	 *
+	 * @return void
+	 * @throws GuzzleException
+	 */
+	public function userCopiesFileFromAndToSpaceBetweenSpaces(
+		string $user,
+		string $fileSource,
+		string $fromSpaceName,
+		string $fileDestination,
+		string $toSpaceName
+	):void {
+		$space = $this->getSpaceByName($user, $fromSpaceName);
+		$headers['Destination'] = $this->destinationHeaderValueWithSpaceName($user, $fileDestination, $toSpaceName);
+		$fullUrl = $space["root"]["webDavUrl"] . '/' . ltrim($fileSource, "/");
+		$this->copyFilesAndFoldersRequest($user, $fullUrl, $headers);
+	}
+
+	/**
+	 * returns a url for destination with spacename
+	 *
+	 * @param string $user
+	 * @param string $fileDestination
+	 * @param string $spaceName
+	 *
+	 * @return string
+	 * @throws GuzzleException
+	 */
+	public function destinationHeaderValueWithSpaceName(string $user, string $fileDestination, string $spaceName):string {
+		$space = $this->getSpaceByName($user, $spaceName);
+		return $space["root"]["webDavUrl"] . '/' . \ltrim($fileDestination, '/');
+	}
+
+	/**
+	 * COPY request for files|folders
+	 *
+	 * @param string $user
+	 * @param string $fullUrl
+	 * @param string $headers
+	 *
+	 * @return void
+	 * @throws GuzzleException
+	 */
+	public function copyFilesAndFoldersRequest(string $user, string $fullUrl, array $headers):void {
```

Comment:
do you mean putting this fn to `WebDavHElper`?

Reply:
```
return WebDavHelper::makeDavRequest(
			$this->getBaseUrl(),
			$user,
			$password,
			$method,
			$path,
			$headers,
			$this->getStepLineRef(),
			$body,
			$davPathVersion,
			$type,
			null,
			"basic",
			$stream,
			$this->httpRequestTimeout,
			null,
			$urlParameter,
			$doDavRequestAsUser
		);
```
If i am not wrong @individual-it  is saying about this function. We have got required parameters to pass and do not need extra work to do to it. May be  sending direct request seems better or ? @kiranparajuli589 @individual-it

Reply:
Hmm, I've looked inside this `WebDavHelper::makeDavRequest` fn. Some limitations to this fn are:
- we cannot provide the full URL, the fn. will calculate the fullUrl according to the davVersions used and the path provided
- if we're using the spaces dav version, it only goes for the Personal space (which is not useful, because here we may be copying to/from different spaces)

we've to do some tricky refactorings to make this work from the `makeDavRequest` fn.